### ISSUE TITLE
Apply 'disabled' styling when `aria-disabled` is set on a `Button` or `IconButton`

### ISSUE THREAD

**Issue by iansan5653** [primer/react#3991]
In many cases it's not desirable to fully disable a button control because it can make that button difficult for some users to locate. A disabled button is not focusable and doesn't respond to interaction at all, and it doesn't have any indication of why it's disabled. 

While it's generally preferable to just avoid disabling buttons and rely on alternative methods instead, in _some_ cases it may be preferable to use `aria-disabled` along with CSS styling to disable a button. This is particularly useful in tandem with a tooltip that appears on focus (this tooltip would be inaccessible on a `disabled` button).

As a component library we should make accessible practices as easy as possible. So I think we shouldn't force consumers to build custom CSS styles in order to use this approach - we should make using `aria-disabled` as easy as `disabled`. With that in mind, I propose that `aria-disabled` should, by default, have the same styling as `disabled` buttons. 

I've focused on buttons here but this actually probably applies to all interactive components that can be disabled.

**Comment by mperrotti** [comment]
@lesliecdubs - if the group agrees with this change, it would fit great in this PR: https://github.com/primer/react/pull/3582

If you prefer we do it in a separate PR, I'm cool with that too.

**Comment by broccolinisoup** [comment]
I think we should re-open this because the button loading state PR was reverted and it is being re-introduced here https://github.com/primer/react/pull/4485 (It is referenced in https://github.com/primer/react/pull/4426 )

**Comment by broccolinisoup** [comment]
@mperrotti I'm assigning this issue to you since you have done the work already in https://github.com/primer/react/pull/4485 but feel free to un-assign yourself if there is anything I misunderstand here.

**Comment by lesliecdubs** [comment]
Should this issue be closed now? I see #4485 was merged.

**Comment by langermank** [comment]
This has been fixed with CSS Modules work on Button 👍
